### PR TITLE
Refactor carousel

### LINKED ISSUE #3394: Refactoring components and stories to use hooks
Tracking the effort for the hooks migration starts here... see the below PRs as references.

### PR DESCRIPTION
**Pull request by shakti97**
#### What does this PR do?
Carousel now using Hooks now. It fixes #3394

#### Where should the reviewer start?
From the Carousel.js file

#### What testing has been done on this PR?
i just run the npm test command
Test Result

#### Screenshots (if appropriate)
here is the video
https://drive.google.com/open?id=1B0ppypWqJgaKmL3A9Lhb4Xl3T9UF8z3E

### CODE REVIEW

File: src/js/components/Carousel/Carousel.js [hunk 1]
```diff
@@ -12,199 +12,169 @@ import { Keyboard } from '../Keyboard';
 import { Stack } from '../Stack';
 import { withFocus } from '../hocs';
 
-class Carousel extends Component {
-  constructor(p) {
-    super(p);
-    this.state = { activeIndex: p.initialChild };
-  }
+const Carousel = props => {
+  const { initialChild, play: playProps } = props;
```
Comment (oorestisime) [inline src/js/components/Carousel/Carousel.js]: can we destructure the props directly in the declaration? would be more in accordance with other components :)

File: src/js/components/Carousel/Carousel.js [hunk 2]
```diff
@@ -12,199 +12,169 @@ import { Keyboard } from '../Keyboard';
...
-    const showSelectors = controls && controls !== 'arrows';
-    const lastIndex = Children.count(children) - 1;
-    const onLeft = activeIndex > 0 ? this.onLeft : undefined;
-    const onRight = activeIndex < lastIndex ? this.onRight : undefined;
-
-    const CurrentIcon = theme.carousel.icons.current;
-    const iconColor = normalizeColor(
-      theme.carousel.icons.color || 'control',
-      theme,
+  const { children, controls, fill, focus, theme, ...rest } = props;
```
Comment (oorestisime) [inline src/js/components/Carousel/Carousel.js]: can we move this destructure as well into the function definition?

File: src/js/components/Carousel/Carousel.js (hunk 1 above)
Comment (shakti97) [reply src/js/components/Carousel/Carousel.js]: okay will do

File: src/js/components/Carousel/Carousel.js (hunk 2 above)
Comment (shakti97) [reply src/js/components/Carousel/Carousel.js]: yes, will do

**Comment by shakti97** [comment]
Destructuring

**Comment by shakti97** [comment]
OKay will check and fix it

**Comment by ShimiSun** [comment]
hi @shakti97, any updates on this PR?

**Comment by ShimiSun** [comment]
hi buddy @shakti97, do you need help with this one?

**Comment by ShimiSun** [comment]
Thanks for contributing! The Carousel was refactored to Function component on another PR effort, I'll be closing this PR.

**Pull request closed without merging**